content/articles/creating-an-animated-javafx-field-validation-alert/index.md

```diff
@@ -0,0 +1,758 @@
+
+```java
+                new animatefx.animation.Shake(usernameTextField).play();
+                new animatefx.animation.Wobble(usersIcon).play();
+                new animatefx.animation.Shake(userPassword).play();
+                new animatefx.animation.Wobble(passwordIcon).play();
+```
+
+This code will add a '**Shake**' animation to the inputs when there is an error. It also adds a '**Wobble**' animation to the images.
+
+- In the procedure that checks if the username is blank, add the following code to it:
+
+```java
+                    new animatefx.animation.Shake(usernameTextField).play();
+                    new animatefx.animation.Pulse(usersIcon).play();
+```
+
+The code adds the '**Pulse**' and '**Shake**' effect to the input and icon respectively.
+
+- In the procedure that checks if the username is blank, add the following code:
+
+```java
+                    new animatefx.animation.Shake(userPassword).play();
+                    new animatefx.animation.Wobble(passwordIcon).play();
+```
+
+The code also does as the previous one but now to the PasswordField and its icon.
+
+- In the procedure that checks if the password is less than four characters, add a slightly gentle effect.
+Add the following code to it:
+
+```java
+                new animatefx.animation.FadeIn(userPassword).play();
+                new animatefx.animation.Wobble(passwordIcon).play();
+```
+
+It adds a '**FadeIn**' and a '**Wobble**' effect to the PasswordField and its icon respectively.
+
+- Finally, in the final `else` statement, add a small `Tada` animation to the `invalidDetails` label.
+That animation shows that the process is successful. The code to add is shown below:
+
+```java
+                new animatefx.animation.Tada(invalidDetails).play();
+```
+
+The final product is as shown in the GIF below:
+
+![Validation Animations](/engineering-education/creating-an-animated-javafx-field-validation-alert/Animations.gif "Validation Animations")
+
+The code for this article can be found in [this](https://github.com/justusmbuvi/Creating-an-Animated-JavaFX-field-Validation-alert) repository.
+
+### Further learning
+
+- Add more animations to the project to tailor it as needed. Choose any animation from the [link](https://github.com/Typhon0/AnimateFX/tree/master/animatefx/src/main/java/animatefx/animation) to see which is best.
+- Add more complex validations by using [ValidatorFX](https://github.com/effad/ValidatorFX) library.
+- Choose more colorful templates for the project from the [Coolor.co](https://coolors.co/palettes/trending) site. The site is free and has some cool templates.
+
+### Conclusion
+
+In the article, the following have been accomplished:
+
+- Create a JavaFX login form
+- Apply styling to the application
+- Create an input validation method
+- Add effects based on the outcomes of the validation
+- Add animations on the controls
+- Learnt more types of animations available for JavaFX projects

```

Comment:
Use complete sentences terminated with a period.

Reply:
Got you!